server-app/src/users/users.service.ts

```diff
@@ -7,20 +7,27 @@ import { IUser } from './interfaces/user.interface';
 import { ICreateUserInput } from './interfaces/create-user-input.interface';
 import { IUpdateUserInput } from './interfaces/update-user-input.interface';
 import { IUpdateUserPartialInput } from './interfaces/update-user-partial-input.interface';
+import { MongooseDatabaseModule } from '../database/mongoose-database.module';
 
 let users = [];
 
 @Injectable()
 export class UsersService {
-  findOneAndUpdate(id: number, updateBody: IUpdateUserPartialInput): IUser {
-    const user = this.findOneById(id);
+  constructor(private readonly db: MongooseDatabaseModule) {
+    this.db.connect();
+  }
+
+  async findOneAndUpdate(
+    id: number,
+    updateBody: IUpdateUserPartialInput,
+  ): Promise<IUser> {
+    const user = await this.findOneById(id);
     return this.updatePartially(user.id, updateBody);
   }
```

Comment:
binding DB connection to this may bring a lot of problems with a production database, better to rework it to use approach that I provided
Also, it is not an abstraction if you directly set the database layer